cpp/src/datetime/datetime_ops.cu

```diff
@@ -290,6 +362,64 @@ std::unique_ptr<column> add_calendrical_months(column_view const& timestamp_colu
   return output;
 }
 
+std::unique_ptr<column> ceil_general(column_view const& column,
```

Comment:
can `ceil_general` be a template function on `datetime_component` so we avoid the switch below?
We know the type as compile time since there is a separate function for each type that calls `ceil_general`.